app/controllers/admin_controller.rb

```diff
@@ -0,0 +1,49 @@
+class AdminController < ApplicationController
+  def map
+  end
+
+  def edit_mark
+    if params[:commit] == "Delete Mark (make pullable)"
```

Comment:
Overall LGTM, except for `params[:commit]`. If it's an input by admin, `"Delete Mark (make pullable)"` seems too long for verification.

Reply:
Fixed! I now use a "delete" variable